### ISSUE TITLE
Wave_write object from wav library confused with Wave_read object

### ISSUE THREAD

**Issue by ksylvan** [microsoft/vscode-pylint#603]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): Python 3.13.2 (Homebrew on MacOS)
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): venv
- Operating system (and version): macOS 15.3.2 (24D81)
- Version of tool extension you are using: ms-python.pylint version 2025.2.0

### Expected Behavior

No linting errors.

### Actual Behavior

![Image](https://github.com/user-attachments/assets/74d22a03-0503-44df-92da-5f04b9d593ac)

```plaintext
Instance of 'Wave_read' has no 'setnchannels' member; maybe 'getnchannels'?
Instance of 'Wave_read' has no 'setsampwidth' member; maybe 'getsampwidth'?
Instance of 'Wave_read' has no 'setframerate' member; maybe 'getframerate'?
Instance of 'Wave_read' has no 'writeframes' member
```

## Reproduction Steps:

Put this in `test.py`:

```python
#!/usr/bin/env python

"""Reproduce Wav_write pylint bug"""

import wave
import sys
import numpy as np

OUTPUT_FILENAME = "/tmp/test.wav"
SAMPLE_RATE = 44100

stereo_int16 = np.empty((0, 2), dtype=np.int16)

# Write the combined audio data to a stereo WAV file
print(f"Writing audio to '{OUTPUT_FILENAME}'...")
try:
    with wave.open(OUTPUT_FILENAME, "wb") as wf:
        wf.setnchannels(2)  # Stereo
        wf.setsampwidth(2)  # 16 bits per sample (2 bytes)
        wf.setframerate(SAMPLE_RATE)
        # Write the frames as bytes
        wf.writeframes(stereo_int16.tobytes())
    print(f"Audio file '{OUTPUT_FILENAME}' generated successfully.")
except (wave.Error, OSError, IOError) as e:
    print(f"Error writing WAV file '{OUTPUT_FILENAME}': {e}", file=sys.stderr)
    sys.exit(1)
```

The code works correctly:

```plaintext
$ ./test.py 
Writing audio to '/tmp/test.wav'...
Audio file '/tmp/test.wav' generated successfully.
```

## Logs:

```json
2025-03-26 16:46:28.736 [info] [Trace - 4:46:28 PM] Received notification 'textDocument/publishDiagnostics'.
2025-03-26 16:46:28.736 [info] Params: {
    "uri": "file:///Users/kayvan/src/binaural/test.py",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 17,
                    "character": 8
                },
                "end": {
                    "line": 17,
                    "character": 23
                }
            },
            "message": "Instance of 'Wave_read' has no 'setnchannels' member; maybe 'getnchannels'?",
            "severity": 1,
            "code": "E1101:no-member",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html"
            },
            "source": "Pylint"
        },
        {
            "range": {
                "start": {
                    "line": 18,
                    "character": 8
                },
                "end": {
                    "line": 18,
                    "character": 23
                }
            },
            "message": "Instance of 'Wave_read' has no 'setsampwidth' member; maybe 'getsampwidth'?",
            "severity": 1,
            "code": "E1101:no-member",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html"
            },
            "source": "Pylint"
        },
        {
            "range": {
                "start": {
                    "line": 19,
                    "character": 8
                },
                "end": {
                    "line": 19,
                    "character": 23
                }
            },
            "message": "Instance of 'Wave_read' has no 'setframerate' member; maybe 'getframerate'?",
            "severity": 1,
            "code": "E1101:no-member",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html"
            },
            "source": "Pylint"
        },
        {
            "range": {
                "start": {
                    "line": 21,
                    "character": 8
                },
                "end": {
                    "line": 21,
                    "character": 22
                }
            },
            "message": "Instance of 'Wave_read' has no 'writeframes' member",
            "severity": 1,
            "code": "E1101:no-member",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html"
            },
            "source": "Pylint"
        }
    ]
}
```

## Outcome When Attempting Debugging Steps:

Did running it from the command line work? No.

**Comment by karthiknadig** [comment]
@ksylvan this repo is for the extension which is a wrapper around `pylint` to expose the features of pylint to VC Code. Please file this on `pylint` the Python library here: https://github.com/pylint-dev/pylint/issues
